sql/auth/sql_authorization.cc

```diff
@@ -4389,9 +4389,11 @@ bool check_fk_parent_table_access(THD *thd,
 
   // Return if engine does not support Foreign key Constraint.
   if (!ha_check_storage_engine_flag(db_type, HTON_SUPPORTS_FOREIGN_KEYS)) {
+    sql_mode_t sql_mode = thd->variables.sql_mode;
+    if(thd->slave_thread) sql_mode = global_system_variables.sql_mode;
```

Comment:
use a = if true ? ... : ..., as next:
```
sql_mode_t sql_mode = thd->slave_thread ? global_system_variables.sql_mode : thd->variables.sql_mode;
```
Please rewrite this logical code in other files.